src/core/fetch/index.js

```diff
@@ -111,7 +112,8 @@ export function Fetch(Base) {
         };
 
         // and a handler that is called if content failed to fetch
-        const contentFailedToFetch = _error => {
+        const contentFailedToFetch = (_error, response) => {
+          this.route.response = response;
           this._fetchFallbackPage(path, qs, cb) || this._fetch404(file, qs, cb);
```

Comment:
we have a ` this._fetch404` to load custom pages here, via config [notfoundpage](https://docsify.js.org/#/configuration?id=notfoundpage).

Shall we keep return 404 pages behaviors, no matter what response status is?
It seem inconsistent behavior when we wanna distinguish response status codes in changes.
Besides, somehow it seems broke the custom 404 page function either when I test this PR locally.

Reply:
@Koooooo-7 @sy-records --

Good catch. This has been fixed. 

The `notFoundPage` configuration option is now handled properly, however it now only applies to 404 errors instead of *all* errors (as it did previously). This is (IMHO) the right thing to do, as both the option name and our documentation indicates that this option *is intended* to only apply to 404 errors.

All non-404 errors will render the default error content which displays the error status and status text (see screenshot above). Users who want to render error-specific content can now do so by checking the `vm.route.response` data and returning custom content via a plugin (see screenshot above).

Reply:
Good job! Thank you!